### PR TITLE
Un-hardcode positions and colors in `2d_shapes` example

### PR DESCRIPTION
**Pull request by rparrett**
# Objective

We recently got some neat new 2d shapes and the shapes are no longer centered on the screen.

The hardcoded positions and colors are a pain to deal with when a new shape is added.

## Solution

Delete a bunch of code and position shapes evenly. Assign colors evenly too.

## Before

Screenshot 2024-02-14 at 3 17 40 PM

## After

Screenshot 2024-02-14 at 3 17 24 PM

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

File: examples/2d/2d_shapes.rs [hunk 1]
```diff
@@ -9,64 +12,41 @@ fn main() {
...
+        Mesh2dHandle(meshes.add(Triangle2d::new(
+            Vec2::Y * 50.0,
+            Vec2::new(-50.0, -50.0),
+            Vec2::new(50.0, -50.0),
+        ))),
+    ];
+    let num_shapes = shapes.len();
+
+    for (i, shape) in shapes.into_iter().enumerate() {
+        let color = Color::hsl(360. * i as f32 / num_shapes as f32, 0.95, 0.7);
```
Comment (BD103) [inline examples/2d/2d_shapes.rs:39]:
```suggestion
        // Distribute colors evenly across the rainbow.
        let color = Color::hsl(360. * i as f32 / num_shapes as f32, 0.95, 0.7);
```

File: examples/2d/2d_shapes.rs [hunk 2]
```diff
@@ -9,64 +12,41 @@ fn main() {
...
+
+        commands.spawn(MaterialMesh2dBundle {
+            mesh: shape,
+            material: materials.add(color),
+            transform: Transform::from_translation(Vec3::new(
+                -X_EXTENT / 2. + i as f32 / (num_shapes - 1) as f32 * X_EXTENT,
```
Comment (BD103) [inline examples/2d/2d_shapes.rs:45]:
```suggestion
                // Distribute shapes from -X_EXTENT to +X_EXTENT.
                -X_EXTENT / 2. + i as f32 / (num_shapes - 1) as f32 * X_EXTENT,
```

**Review by BD103 — APPROVED** [review]
Looks good, especially the color changes! I just suggested a few places where comments might be appreciated (non-blocking, though). Feel free to reword them however you like.

File: examples/2d/2d_shapes.rs [hunk 3]
```diff
@@ -9,64 +12,43 @@ fn main() {
...
+
+        commands.spawn(MaterialMesh2dBundle {
+            mesh: shape,
+            material: materials.add(color),
+            transform: Transform::from_translation(Vec3::new(
+                // Distribute shapes from -X_EXTENT to +X_EXTENT.
+                -X_EXTENT / 2. + i as f32 / (num_shapes - 1) as f32 * X_EXTENT,
+                0.0,
+                0.0,
+            )),
```
Comment (Jondolf) [inline examples/2d/2d_shapes.rs:50]:
```suggestion
            transform: Transform::from_xyz(
                // Distribute shapes from -X_EXTENT to +X_EXTENT.
                -X_EXTENT / 2. + i as f32 / (num_shapes - 1) as f32 * X_EXTENT,
                0.0,
                0.0,
            )),
```

**Review by Jondolf — APPROVED** [review]
LGTM! Left a tiny non-blocking suggestion

File: examples/2d/2d_shapes.rs [hunk 4]
```diff
@@ -9,64 +12,43 @@ fn main() {
...
+            mesh: shape,
+            material: materials.add(color),
+            transform: Transform::from_xyz(
+                // Distribute shapes from -X_EXTENT to +X_EXTENT.
+                -X_EXTENT / 2. + i as f32 / (num_shapes - 1) as f32 * X_EXTENT,
+                0.0,
+                0.0,
+            )),
+            ..default()
+        });
```
Comment (Jondolf) [inline examples/2d/2d_shapes.rs:52]: Whoops, my bad, the closing delimiter needs to be removed too of course 🙈
```suggestion
            transform: Transform::from_xyz(
                // Distribute shapes from -X_EXTENT to +X_EXTENT.
                -X_EXTENT / 2. + i as f32 / (num_shapes - 1) as f32 * X_EXTENT,
                0.0,
                0.0,
            ),
            ..default()
        });
```

**Pull request merged**